tests/end-to-end.rs

```diff
@@ -161,28 +167,54 @@ where
 
 
 /// Serve our registry.
-fn serve_registry() -> (JoinHandle<()>, PathBuf, SocketAddr) {
-  let root = tempdir().unwrap();
-  let path = root.path().to_owned();
+fn serve_registry(root_path: RegistryRootPath) -> (JoinHandle<()>, PathBuf, SocketAddr) {
+  let path = match root_path {
+    RegistryRootPath::Absolute => tempdir().unwrap().path().to_owned(),
+    RegistryRootPath::Relative => {
+      let root = tempdir().unwrap().into_path();
```

Comment:
So now we are leaking temporary directories all over the place. Not okay.

Reply:
Good catch, it's now fixed